lib/rouge/lexers/idris.rb

```diff
@@ -0,0 +1,206 @@
+        do if then else case in
+      )
+
+      ascii = %w(
+        NUL SOH [SE]TX EOT ENQ ACK BEL BS HT LF VT FF CR S[OI] DLE
+        DC[1-4] NAK SYN ETB CAN EM SUB ESC [FGRU]S SP DEL [abfnrtv\\\"'\&]
+      )
+
+      prelude_function = %w(
+        abs acos all and any asin atan atan2 break ceiling compare concat
+        concatMap const cos cosh curry cycle div drop dropWhile elem
+        encodeFloat enumFrom enumFromThen enumFromThenTo enumFromTo exp
+        fail filter flip floor foldl foldl1 foldr foldr1 fromInteger fst
+        gcd getChar getLine head id init iterate last lcm length lines log
+        lookup map max maxBound maximum maybe min minBound minimum mod
+        negate not null or pi pred print product putChar putStr putStrLn
+        readFile recip repeat replicate return reverse scanl scanl1 sequence
+        sequence_ show sin sinh snd span splitAt sqrt succ sum tail take
+        takeWhile tan tanh uncurry unlines unwords unzip unzip3 words
+        writeFile zip zip3 zipWith zipWith3
+      )
+
+      state :basic do
+        rule %r/\s+/m, Text
+        rule %r/{-#/, Comment::Preproc, :comment_preproc
+        rule %r/{-/, Comment::Multiline, :comment
+        rule %r/^\|\|\|.*$/, Comment::Doc
+        rule %r/--(?![!#\$\%&*+.\/<=>?@\^\|_~]).*?$/, Comment::Single
+      end
+
+      # nested commenting
+      state :comment do
+        rule %r/-}/, Comment::Multiline, :pop!
+        rule %r/{-/, Comment::Multiline, :comment
+        rule %r/[^-{}]+/, Comment::Multiline
+        rule %r/[-{}]/, Comment::Multiline
+      end
+      state :comment_preproc do
+        rule %r/-}/, Comment::Preproc, :pop!
+        rule %r/{-/, Comment::Preproc, :comment
+        rule %r/[^-{}]+/, Comment::Preproc
+        rule %r/[-{}]/, Comment::Preproc
+      end
+
+      state :directive do
+        rule %r/\%(default)\s+(total|partial)/, Keyword  # totality
+        rule %r/\%(access)\s+(public|abstract|private|export)/, Keyword  # export
+        rule %r/\%(language)\s+(.*)/, Keyword  # language
+        rule %r/\%(provide)\s+.*\s+(with)\s+/, Keyword  # type
+      end
+
+      state :prelude do
+        rule %r/\b(Type|Exists|World|IO|IntTy|FTy|File|Mode|Dec|Bool|Ordering|Either|IsJust|List|Maybe|Nat|Stream|StrM|Not|Lazy|Inf)\s/, Keyword::Type
+        rule %r/\b(Eq|Ord|Num|MinBound|MaxBound|Integral|Applicative|Alternative|Cast|Foldable|Functor|Monad|Traversable|Uninhabited|Semigroup|Monoid)\s/, Name::Class
+        rule %r/\b(?:#{prelude_function.join('|')})[ ]+(?![=:-])/, Name::Builtin
+      end
+
+      state :root do
+        mixin :basic
+        mixin :directive
+
+        rule %r/\bimport\b/, Keyword::Reserved, :import
+        rule %r/\bmodule\b/, Keyword::Reserved, :module
+        rule %r/\b(?:#{reserved.join('|')})\b/, Keyword::Reserved
+        rule %r/\b(Just|Nothing|Left|Right|True|False|LT|LTE|EQ|GT|GTE)\b/, Keyword::Constant
+        # function signature
+        rule %r/^([\w']+)\s*(:)/, Name::Function
```

Comment:
Is there any reason to use capture groups for `[\w']` and `:`? Is it just for legibility?

Reply:
Addressed in https://github.com/rouge-ruby/rouge/pull/1464/commits/3a31bf6eb5dd8193f34b2de9cefea6d06d2768e3